### PR TITLE
Add Encoder map support for rart

### PR DESCRIPTION
**Pull request by alabahuy**
## Description

- Add encoder mapping for rart4x4, rart75, rart75hs, rart75m and rartpad
- Add encoder map enable to utilize encoder mapping
- Fix pin capslock and RGB fot rart80

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rart/rart75hs/rart75hs.c [hunk 1]
```diff
@@ -12,20 +12,4 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-#include "rart75hs.h"
```
Comment (drashna) [inline keyboards/rart/rart75hs/rart75hs.c:15]: This should be left.  It's useful in the case that somebody is compiling from the configurator.

File: keyboards/rart/rart75m/rart75m.c [hunk 2]
```diff
@@ -14,22 +14,6 @@
  */
 #include "rart75m.h"
 
-#ifdef ENCODER_ENABLE
```
Comment (drashna) [inline keyboards/rart/rart75m/rart75m.c:17]: Same

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/rart/rart75hs/rart75hs.c (hunk 1 above)
Comment (alabahuy) [reply keyboards/rart/rart75hs/rart75hs.c]: ah i see, Allrights @drashna :)

File: keyboards/rart/rart75m/rart75m.c (hunk 2 above)
Comment (alabahuy) [reply keyboards/rart/rart75m/rart75m.c]: Done :)

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**